core/src/main/java/org/bitcoinj/script/ScriptBuilder.java

```diff
@@ -119,6 +123,65 @@ public ScriptBuilder smallNum(int index, int num) {
         return addChunk(index, new ScriptChunk(Script.encodeToOpN(num), null));
     }
 
+    /** Adds the given number as a push data chunk to the given index in the program. */
+    public ScriptBuilder bigNum(int index, long num) {
+        final byte[] data;
+
+        // This encoding is copied directly from
+        // https://github.com/bitcoin/bitcoin/blob/b48da5c1894a70f8fa2a50deb2e056c38ed27ecb/src/script/script.h#L303
+        // Specifically note that the encoded format is not always the minimum
+        // possible.
+        if (num == 0) {
+            data = new byte[0];
+        } else {
+            Stack<Byte> result = new Stack<Byte>();
+            final boolean neg = num < 0;
+            long absvalue = Math.abs(num);
+
+            while (absvalue != 0)
+            {
+                result.push((byte) (absvalue & 0xff));
+                absvalue >>= 8;
+            }
+
+            //    - If the most significant byte is >= 0x80 and the value is positive, push a
```

Comment:
Mostly because those notes are copied directly from the C++ code. Cleaning them up now and will re-push.